lib/sequenceserver/job_remover.rb

```diff
@@ -0,0 +1,59 @@
+require 'sequenceserver/job.rb'
+
+module SequenceServer 
+  class JobRemover
+
+    DEFAULT_LIFETIME = 86400
+
+    def initialize(lifetime)
+      parse lifetime
+      spawn_cleanup if @clean_needed
+    end
+
+    def spawn_cleanup
+      Thread.abort_on_exception = true if ENV['RACK_ENV'] == 'development'
+
+      Thread.new do 
+        loop do
+          #leave incomplete jobs
+          current_jobs = Job.all.select{ |f| !f.completed_at.nil? }
+          oldest_time = Time.at(0)
+
+          unless current_jobs.empty?
+            oldest_time = current_jobs.max do |left, right|
+              left.completed_at <=> right.completed_at
+            end.completed_at
+ 
+            exp_jobs = current_jobs.select do |job|
+              (job.completed_at + @lifetime) < Time.now
+            end
+          end
+
+          next_cleanup = @lifetime
```

Comment:
Can you set this as an instance variable? That will make it possible to query job cleaner's state from outside, for debugging, instrumenting, etc.